custom_components/spotcast/helpers.py

```diff
@@ -60,3 +63,43 @@ async def run(*args, loop=None, executor=None, **kwargs):
         return await loop.run_in_executor(executor, pfunc)
 
     return run
+
+def get_uri_from_search(search, token):
+
+    results = __get_search_results(search, token)
+
+    return results[0]['name']
+
+def __get_search_results(search, token):
+    BASE_URL = "https://api.spotify.com/v1/search?q="
```

Comment:
Instead of doing all this manual api logic, I think you can utilize the spotipy client

https://github.com/fondberg/spotcast/blob/e2528b7ffd8e61e37e5a15a03452b6fed11c38aa/custom_components/spotcast/spotcast_controller.py#L175

Reply:
I'll check that, I basically reused my prototype that I built to fiddle with the api, but if its already implemented obviously we are better off centralising.

Reply:
Made the requested changes